### PR TITLE
soc: mcxn94x: Add HWINFO and fix USB_Next support

### LINKED ISSUE #82755: drivers: usb: udc: mcux_ehci: UsageFault when enabling device_next
**Describe the bug**

UsageFault while using `device_next` driver `nxp,ehci`

**To Reproduce**
```
$ west build -b frdm_mcxn947/mcxn947/cpu0 samples/subsys/usb/cdc_acm -- -DCONF_FILE=usbd_next_prj.conf
$ west debug
(gdb) break USB_DeviceEhciInit
(gdb) continue
(gdb) stepi 20
```

**Expected behavior**

`/dev/ttyACM0` enumerating at USB2 HighSpeed.

**Impact**

The upcoming `device_next` USB stack is not functional.
Not possible to test CDC ACM (and UVC) on this very nice board yet.
Hobby/procrastination project only, no production at halt.

**Logs and console output**

```
Processor connection reset
(gdb) break USB_DeviceEhciInit
Breakpoint 1 at 0x10007a54: file /home/tinyvision/zephyrproject/modules/hal/nxp/mcux/middleware/mcux-sdk-middleware-usb/device/usb_device_ehci.c, line 1469.
Note: automatically using hardware breakpoints for read-only addresses.
(gdb) continue
Continuing.

Breakpoint 1, USB_DeviceEhciInit (controllerId=2 '\002', handle=0x30001f68 <priv_data_0+4>, ehciHandle=0x30001f68 <priv_data_0+4>) at /home/tinyvision/zephyrproject/modules/hal/nxp/mcux/middleware/mcux-sdk-middleware-usb/device/usb_device_ehci.c:1469
1469        if ((controllerId < (uint8_t)kUSB_ControllerEhci0) ||
(gdb) stepi 20

Program received signal SIGSTOP, Stopped (signal).
z_arm_usage_fault () at /home/tinyvision/zephyrproject/zephyr/arch/arm/core/cortex_m/fault_s.S:80
80              mrs r0, MSP
(gdb)
```

**Environment (please complete the following information):**

 - OS: `Linux vm1 6.8.0-49-generic #49-Ubuntu SMP PREEMPT_DYNAMIC Mon Nov  4 02:06:24 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux`
 - Toolchain: Zephyr SDK `0.16.8`
 - Commit SHA Zephyr `2cbe868f002d08b5227c5fdda72ef070aa7984b7`, HAL NXP `d291bdcc4a59bace5ae7453e777e06080ccda8ce`

**Additional context**

The same instruction succeeds while using `device`, but not when using `device_next`.
Feel free to point me at anything you'd like me to test.

Something that raised my curiosity:

```
(gdb) print ehciState->registerBase->USBCMD
Cannot access memory at address 0x5010b140
(gdb)
```

### PR DESCRIPTION
**Pull request by mmahadevan108**
1. Add HWINFO support by reading the UUID from
2. Fix USB next support on this FRDM-MCXN947 board

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/82755

### CODE REVIEW

File: dts/arm/nxp/nxp_mcxn94x_common.dtsi [hunk 1]
```diff
@@ -582,6 +582,11 @@
 			/* MCXN94x ROM Flash API supports writing of 128B pages. */
 			write-block-size = <128>;
 		};
+
+		uuid: flash@1100000 {
```
Comment (decsny) [inline dts/arm/nxp/nxp_mcxn94x_common.dtsi:586]: node name says flash?

Comment (mmahadevan108) [reply dts/arm/nxp/nxp_mcxn94x_common.dtsi:586]: The uuid is in the flash. Similar to some LPC devices like LPC55S69.

Comment (decsny) [reply dts/arm/nxp/nxp_mcxn94x_common.dtsi:586]: I don't think the uuid is erasable, which is a defining feature of a flash memory, that's what the name comes from, so don't think the node name flash makes sense here

Comment (mmahadevan108) [reply dts/arm/nxp/nxp_mcxn94x_common.dtsi]: I have updated the node name.

**Review by decsny — APPROVED** [review]

**Review by EmilioCBen — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**